streaming/index.js

```diff
@@ -731,34 +768,56 @@ const startServer = async () => {
             });
           }
 
-          // Check filters
-          if (req.cachedFilters && !payload.filtered) {
-            const mutatedPayload = { ...payload };
+          // Apply cachedFilters against the payload, constructing a
+          // `filter_results` array of FilterResult entities
+          if (req.cachedFilters) {
```

Comment:
This can just be `if (req.cachedFilters)` because on line 713, we return early if the `payload` has already been filtered by the Ruby on Rails side (meaning `payload.filtered` is present)